pkg/k8shandler/collection.go

```diff
@@ -94,8 +94,8 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateCollection(proxyConfi
 
 			// remove our finalizer from the list and update it.
 			collectorServiceAccount.ObjectMeta.Finalizers = utils.RemoveString(collectorServiceAccount.ObjectMeta.Finalizers, metav1.FinalizerDeleteDependents)
-			err = clusterRequest.Create(collectorServiceAccount)
-			if err != nil && !errors.IsAlreadyExists(err) {
+			if err = clusterRequest.Update(collectorServiceAccount); err != nil {
+				logger.Warnf("Unable to update the collector serviceaccount %q finalizers: %v", collectorServiceAccount.Name, err)
```

Comment:
do we know what this block is actually trying to accomplish? if it wasn't correct before and fluentd worked, do we even need any of this at all?

Reply:
I believe this was originally added by @nhosoi to address a bug where fluentd was not being terminated in a timely fashion and it was related to the finalizers.  We would need to investigate the commit hash to find the bug.